apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/AppConfiguration.java

```diff
@@ -288,18 +275,18 @@ static AppConfiguration readConfigFromEnvironmentVariables() {
           e);
     }
 
-    return new AppConfiguration(
-        new ExtractionOptions(s3BucketName, allowedRifFileType),
+    DatabaseOptions databaseOptions =
+        new DatabaseOptions(databaseUrl, databaseUsername, databasePassword.toCharArray());
+    ExtractionOptions extractionOptions = new ExtractionOptions(s3BucketName, allowedRifFileType);
+    LoadAppOptions loadOptions =
         new LoadAppOptions(
+            databaseOptions,
             hicnHashIterations,
             hicnHashPepper,
-            databaseUrl,
-            databaseUsername,
-            databasePassword.toCharArray(),
             loaderThreads,
-            idempotencyRequired.get().booleanValue(),
-            fixupsEnabled,
-            fixupThreads));
+            idempotencyRequired.get().booleanValue());
+    CcwRifLoadOptions ccwRifLoadOptions = new CcwRifLoadOptions(extractionOptions, loadOptions);
```

Comment:
Loading all of the configuration variables here creates a very tight dependency between the app package and the job package implementations.  It would be nice if we could delegate configuration loading to the individual jobs rather than centralize it here.  Perhaps by passing a BiFunction<String,String,String> or some equivalent interface of our own to their constructor to abstract config setting lookup so we could choose between environment variables or java properties later without changing the submodule configs.

Reply:
Yeah, I was thinking about that myself; it's a good insight. I opted to not address it for now, but I expect it'll be one of those things we come back to in a future PR.